app/main.py

```diff
@@ -1,25 +1,89 @@
+from __future__ import annotations
 from abc import ABC
+from typing import Type
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: Type[ChildrenSlideLimitationValidator
+                        | AdultSlideLimitationValidator],
+            name: str) -> None:
+        self.public_name = "_" + name
+
+    def __get__(
+            self,
+            instance: (ChildrenSlideLimitationValidator
+                       | AdultSlideLimitationValidator),
+            owner: Type[ChildrenSlideLimitationValidator
+                        | AdultSlideLimitationValidator],
+    ) -> int:
+        return getattr(instance, self.public_name)
+
+    def __set__(
+            self,
+            instance: (ChildrenSlideLimitationValidator
+                       | AdultSlideLimitationValidator),
+            value: int
+    ) -> None:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.public_name, value)
+        else:
+            setattr(instance, self.public_name, None)
```

Comment:
better to raise exeption

Reply:
Done. And changed solution for method "can_access", because tests failed